Tasks/AzureVmssDeployment/operations/VirtualMachineScaleSet.ts

```diff
@@ -137,46 +137,56 @@ export default class VirtualMachineScaleSet {
+                compress.createArchive(this.taskParameters.customScriptsDirectory, archive.compression, archive.filePath);
 
-            // copy invoker script to same dir as archive
-            tl.cp(invokerScriptPath, archive.directory, "-f", false);
-            console.log(tl.loc("CopiedInvokerScript", archive.directory));
+                let invokerScriptPath: string;
+                let invokerCommand: string;
 
-            console.log(tl.loc("CustomScriptsArchiveFile", archive.filePath));
-            tl.debug("Invoker command: " + invokerCommand);
-            return <CustomScriptsInfo>{
-                localDirPath: archive.directory,
-                command: invokerCommand
-            };
+                if(osType === "Windows") {
 
-        } catch (error) {
-            tl.warning(tl.loc("CustomScriptsArchivingFailed") + " Error: " + error);
-            return <CustomScriptsInfo>{
-                localDirPath: this.taskParameters.customScriptsPath,
-                command: this.taskParameters.customScriptCommand
-            };
+                    let escapedScript = this.taskParameters.customScript.replace(/'/g, "''");
+                    let escapedArgs = this.taskParameters.customScriptArguments.replace(/'/g, "''").replace(/"/g, '"""');
+                    invokerScriptPath = path.join(__dirname, "..", "Resources", "customScriptInvoker.ps1");
+                    invokerCommand = `powershell ./customScriptInvoker.ps1 -zipName '${archive.fileName}' -script '${escapedScript}' -scriptArgs '${escapedArgs}'`;
+                } else {
+                    let escapedScript = this.taskParameters.customScript.replace(/'/g, "'\"'\"'");
+                    let escapedArgs = this.taskParameters.customScriptArguments.replace(/'/g, "'\"'\"'");
+                    invokerScriptPath = path.join(__dirname, "..", "Resources", "customScriptInvoker.sh");
+                    invokerCommand = `./customScriptInvoker.sh '${archive.fileName}' '${escapedScript}' '${escapedArgs}'`;
+                }
+
+                // copy invoker script to same dir as archive
+                tl.cp(invokerScriptPath, archive.directory, "-f", false);
+                console.log(tl.loc("CopiedInvokerScript", archive.directory));
+
+                console.log(tl.loc("CustomScriptsArchiveFile", archive.filePath));
+                tl.debug("Invoker command: " + invokerCommand);
+                return <CustomScriptsInfo>{
+                    localDirPath: archive.directory,
+                    command: invokerCommand
+                };
+
+            } catch (error) {
+                tl.warning(tl.loc("CustomScriptsArchivingFailed") + " Error: " + error);
+            }
+        } else {
+            console.log(tl.loc("SkippedArchivingCustomScripts"));
         }
+
+        return <CustomScriptsInfo>{
+            localDirPath: this.taskParameters.customScriptsDirectory,
+            command: this.taskParameters.customScript
```

Comment:
are you missing arguments here

Reply:
using customScriptInvoker script even when archiving is off. this will make it consistent with archiving flow as well.